src/main/kotlin/no/nav/syfo/syfosmvarsel/varselutsending/VarselProducer.kt

```diff
@@ -8,13 +10,20 @@ import org.apache.kafka.clients.producer.KafkaProducer
 import org.apache.kafka.clients.producer.ProducerRecord
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
+import java.io.IOException
 
 class VarselProducer(private val diskresjonskodeService: DiskresjonskodePortType, private val kafkaproducer: KafkaProducer<String, OppgaveVarsel>, private val oppgavevarselTopic: String) {
     private val log: Logger = LoggerFactory.getLogger("no.nav.syfo.syfosmvarsel")
 
-    fun sendVarsel(oppgaveVarsel: OppgaveVarsel) {
+    suspend fun sendVarsel(oppgaveVarsel: OppgaveVarsel) {
         try {
-            val diskresjonskode: String? = diskresjonskodeService.hentDiskresjonskode(WSHentDiskresjonskodeRequest().withIdent(oppgaveVarsel.mottaker)).diskresjonskode
+            val diskresjonskode: String? = retry(
+                    callName = "hent_diskresjonskode",
+                    retryIntervals = arrayOf(500L, 1000L, 3000L, 5000L, 10000L, 60000L),
+                    legalExceptions = *arrayOf(IOException::class, WstxException::class)
```

Comment:
Er du sikker på at det er disse feilene du får her..? Hvis ikke, skulle vi ha kjørt en try/catch som fanger alle feil, logger dem, og kaster en IOException slik at de fanges av retry, tilsvarende som jeg gjorde for syfosmregler? :)

Reply:
Veit vi får SOAPFaultException, men tenker vi bare ser i prod hvilke som oppstår og så legger vi dei evt til